tokio/src/runtime/time/mod.rs

```diff
@@ -171,15 +169,24 @@ impl Driver {
 
     fn park_internal(&mut self, rt_handle: &driver::Handle, limit: Option<Duration>) {
         let handle = rt_handle.time();
-        let mut lock = handle.inner.state.lock();
-
         assert!(!handle.is_shutdown());
 
-        let next_wake = lock.wheel.next_expiration_time();
-        lock.next_wake =
-            next_wake.map(|t| NonZeroU64::new(t).unwrap_or_else(|| NonZeroU64::new(1).unwrap()));
+        // Finds out the min expiration time to park.
+        let mut next_wake: Option<u64> = None;
+        for id in 0..rt_handle.time().inner.get_shard_size() {
+            let lock = unsafe { rt_handle.time().inner.lock_sharded_wheel(id) };
```

Comment:
Sorry, not sure why `unsafe` was used here. But consider this.
```rust
        let next_wake = (0..handle.inner.get_shard_size())
            .filter_map(|id| handle.inner.lock_sharded_wheel(id).next_expiration_time())
            .fold(None::<u64>, |acc, expiration_time| {
                Some(match acc {
                    Some(t) => t.min(expiration_time),
                    None => expiration_time,
                })
            });
```

Reply:
Thanks. The `unsafe` here is a mistake.